src/plus/launchpad/launchpad.ts

```diff
@@ -630,6 +614,15 @@ export class LaunchpadCommand extends QuickCommand<State> {
 					}
 				}
 
+				// wait a little bit because the active quickpick is updated with some delay
+				await new Promise(resolve => setTimeout(resolve, 100));
+				const hasActiveLaunchpadItems = quickpick.activeItems.find(i => 'item' in i);
+				if (hasActiveLaunchpadItems) {
+					// We have an active item, so we can exit now without sending any requests to API
+					this.updateItemsDebouncer.cancel();
+					return true;
+				}
+
```

Comment:
Could this be handled by the [`onDidChangeActive`](https://code.visualstudio.com/api/references/vscode-api#QuickPick:~:text=onDidChangeActive%3A%20Event%3Creadonly%20T%5B%5D%3E) event?

```typescript
onDidChangeActive: async quickpick => {
	// Not sure if this needs an await
	const hasActiveLaunchpadItems = await quickpick.activeItems.find(i => 'item' in i);
	if (hasActiveLaunchpadItems) {
		this.updateItemsDebouncer.cancel();
		return true;
	}
},
```

Reply:
@sergiolms 
There are following problems:
1. I'm not sure that it's guaranteed that `onDidChangeActive` is called after than `onDidChangeValue`.
2. If after typing, then `onDidChangeValue` won't probably be called at all.
3. It just never gets executed:
<img width="600" src="https://github.com/user-attachments/assets/9baa3530-3445-4c33-8324-5b0b1129bd62">

Reply:
Please take a look at the code suggestion here https://github.com/gitkraken/vscode-gitlens/pull/3770#issuecomment-2485910711 and test if it works for you too. This is working for me:

https://github.com/user-attachments/assets/10479abc-f4e1-401b-8c8c-6717cb0d81f6